hngprojects/hng_boilerplate_nestjs
# feat: Implement reset password endpoint

### What does the PR do?

This PR introduces a new feature for sending password reset emails. It includes an endpoint to handle password reset requests and a service method to send the email with a reset link.

### How should this be manually tested?

1. Send a POST request to /api/v1/auth/password-reset-email with a request body containing:
2. Input the email
```
{
  "email": "test@example.com"
}
```
3. Check the email inbox for the user with the provided email address. 
You should receive an email with a password reset link. check the spam inbox  if not in the inbox

### Checklist of what you did

- [ ] Implemented the endpoint to handle password reset email requests
- [ ] Validated the request body and returned appropriate error messages when validation fails
- [ ] Checked if the user exists before attempting to send the reset email
- [ ] Generated a password reset token and included it in the email link
- [ ] Used the MailerService to send the reset password email
- [ ] Added unit tests for the password reset email functionality, including scenarios for valid and invalid user requests

**Reference Issue**
[FEAT]: [Implement Password Reset Email Functionality](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/303)
